frontend/web/components/import-export/FeatureImport.tsx

```diff
@@ -0,0 +1,303 @@
+        null,
+        page,
+        {
+          search,
+          tags: tags?.length ? tags : undefined,
+        },
+      )
+    }
+  }, [previewEnvironment, tags, search, projectId, page])
+
+  const [createImport, { error, isLoading }] =
+    useCreateFlagsmithProjectImportMutation({})
+
+  const onSubmit = () => {
+    if (env && file) {
+      createImport({
+        environment_id: env.id,
+        file,
+        strategy,
+      }).then((res) => {
+        if (res?.data) {
+          toast('Your import is processing')
+        } else {
+          toast('Failed to import flags')
+        }
+      })
+    }
+  }
+
+  const [strategy, setStrategy] = useState<ImportStrategy>('SKIP')
+
+  useEffect(() => {
+    AppActions.getFeatures(projectId, environment, true, null, null, page, {
+      search,
+      tags: tags?.length ? tags : undefined,
+    })
+  }, [projectId, tags])
+  const {
+    featureStates,
+    projectFlags,
+  }: {
+    projectFlags: ProjectFlag[] | null
+    featureStates: FeatureState[] | null
+  } = useMemo(() => {
+    if (fileData) {
+      const createdDate = new Date().toISOString()
+      const existingFlags: ProjectFlag[] =
+        !!fileData &&
+        !!currentFeatureStates &&
+        currentProjectflags?.map((projectFlag, i) => {
+          const featureState = currentFeatureStates[projectFlag.id]
+          const importItem = fileData.find((v) => v.name === projectFlag.name)
+          let newValue = featureState.value
+          let newEnabled = featureState.enabled
+          if (strategy === 'OVERWRITE_DESTRUCTIVE' && importItem) {
+            if (environment === previewEnvironment) {
+              newEnabled = importItem.enabled
+              newValue = importItem.value
+            } else {
+              newEnabled = importItem.default_enabled
+              newValue = importItem.initial_value
+            }
+          }
+          return {
+            ...projectFlag,
+            created_date: projectFlag.created_date,
+            default_enabled: newEnabled,
+            id: i,
+            initial_value: newValue,
+          }
+        })
+
+      const projectFlags = existingFlags.concat(
+        fileData
+          .filter((v) => {
+            return !existingFlags.find((flag) => flag.name === v.name)
+          })
+          .map(function (importItem, i) {
+            return {
+              created_date: createdDate,
+              default_enabled: importItem.enabled,
+              id: i,
+              initial_value: importItem.value,
+              is_archived: false,
+              is_server_key_only: importItem.is_server_key_only,
+              multivariate_options: importItem.multivariate,
+              name: importItem.name,
+              num_identity_overrides: 0,
+              num_segment_overrides: 0,
+              owner_groups: [],
+              owners: [],
+              project: ProjectStore.model!.id,
+              tags: [],
+              type: 'STANDARD',
```

Comment:
It's not always `STANDARD`, right? I think the multivariate option could come in here.

Reply:
Hmm that's a good point, I guess it's 


```
   type: importItem.multivariate?.length
                ? 'MULTIVARIATE'
                : 'STANDARD',
```